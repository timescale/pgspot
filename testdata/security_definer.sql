-- unsafe sec definer with no search_path
CREATE FUNCTION unsafe_sec_definer2() RETURNS TEXT LANGUAGE SQL SECURITY DEFINER AS $$ SELECT now(); $$;
-- same function but with security invoker should only give warning
CREATE FUNCTION safe_sec_invoker4() RETURNS TEXT LANGUAGE SQL SECURITY INVOKER AS $$ SELECT pg_catalog.now(); $$;
-- unsafe plpgsql sec definer with no search_path
CREATE FUNCTION unsafe_sec_definer6() RETURNS TEXT LANGUAGE PLPGSQL SECURITY DEFINER AS $$ BEGIN RETURN pg_catalog.now(); END; $$;
-- same function but with security invoker should only give warning
CREATE FUNCTION safe_sec_definer8() RETURNS TEXT LANGUAGE PLPGSQL SECURITY INVOKER AS $$ BEGIN RETURN pg_catalog.now(); END; $$;
-- unsafe plpgsql sec definer procedure with no search_path
CREATE PROCEDURE unsafe_sec_definer10() LANGUAGE PLPGSQL SECURITY DEFINER AS $$ BEGIN PERFORM pg_catalog.now(); END; $$;
-- security definer function in C without search_path
CREATE FUNCTION c_sec_definer12() RETURNS TEXT LANGUAGE C SECURITY DEFINER AS 'c_sec_definer12';
