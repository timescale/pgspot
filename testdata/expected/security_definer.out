PS003: SECURITY DEFINER function without explicit search_path: unsafe_sec_definer2() at line 1
PS016: Unqualified function call: now at line 1
PS005: Function without explicit search_path: safe_sec_invoker4() at line 3
PS003: SECURITY DEFINER function without explicit search_path: unsafe_sec_definer6() at line 5
PS005: Function without explicit search_path: safe_sec_definer8() at line 7
PS003: SECURITY DEFINER function without explicit search_path: unsafe_sec_definer10() at line 9

Errors: 3 Warnings: 3 Unknown: 0
